components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/model/IdentityEventListener.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2005,2006 WSO2, Inc. http://www.wso2.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.carbon.identity.core.model;
+
+public class IdentityEventListener {
+
+    private String type;
+    private String name;
+    private IdentityEventListenerProperty property;
+
+    public IdentityEventListener(String type, String name, IdentityEventListenerProperty property) {
+        this.type = type;
+        this.name = name;
+        this.property = property;
+    }
```

Comment:
Shall we define a new object called IdentityEventListenerConfigKey and have 'type' and 'name' as instance variables in it and override the equals() and hashcode() methods ?
Then no need to have IdentityEventListenerProperty class. IdentityEventListener class will itself have IdentityEventListenerConfigKey, enabled and order instance variables.
eventListenerConfiguration will be of type Map<IdentityEventListenerConfigKey, IdentityEventListener>.
This way we can avoid appending type and name with "." also.